### PR TITLE
Added string functions and Dynamic Memory Allocation

### PR DESCRIPTION
**Pull request by vaibhavchaubey**
I have added various string functions and the topic Dynamic Memory Allocation.

### CODE REVIEW

**Comment by vaibhavchaubey** [comment]
@onecompiler-ops please add the HACKTOBERFEST label.

File: c.md [hunk 1]
```diff
@@ -178,9 +178,42 @@ Identifiers are user defined names for variables, functions and arrays.
 
 Strings are an array of characters ended with null character. Enclosed in double quotes.
 
+Declaration
 ```c
     char str[]="onecompiler";
 ```
+### String Functions
```
Comment (onecompiler-ops) [inline c.md:185]: Can you add these as a Markdown table to represent concisely

**Review by onecompiler-ops — CHANGES_REQUESTED** [review]
please use tables where ever possible

**Review by vaibhavchaubey — COMMENTED** [review]
@onecompiler-ops changes done

**Pull request merged**